### PR TITLE
Update v0.6.0

### PR DESCRIPTION
**Pull request by zmccaw-insitro**
* Moved the counting function into C++ to improve performance, and added counts of the contributing variants and carriers. 
* Created a custom class to store and print the output of the main `COAST` function more clearly. 
* Checked updated package passes `R CMD CHECK --as-cran`.

### CODE REVIEW

File: R/allelic_series_test.R [hunk 1]
```diff
@@ -461,44 +459,55 @@ COAST <- function(
...
+  pvals = c(p_burden, p_skat)
   omni_weights <- c(rep(1, n_burden), rep(n_burden / n_skat, n_skat))
 
-  p_omni <- RNOmni::OmniP(p = p_val, w = omni_weights)
+  p_omni <- RNOmni::OmniP(p = pvals, w = omni_weights)
+  pvals <- c(pvals, omni = p_omni)
 
-  # Output.
+  # Only omnibus p-value requested.
   if (return_omni_only) {
```
Comment (ctk3b) [inline R/allelic_series_test.R:491]: n.b. any reason not to move this up above the `pvals <- c(...)` line and return one step earlier?

File: tests/testthat/test-counts.R [hunk 2]
```diff
@@ -11,34 +11,24 @@ test_that("Check variant counter.", {
...
-  # MAC filter of 100: count alleles.
-  obs <- CountAlleles(anno, geno, count_carriers = FALSE, min_mac = 100)
-  exp <- c(n_bmv = 0, n_dmv = 0, n_ptv = 0)
-  expect_equal(obs, exp)
-  
-  # MAC filter of 100: count carriers.
-  obs <- CountAlleles(anno, geno, count_carriers = TRUE, min_mac = 100)
-  exp <- c(n_bmv = 0, n_dmv = 0, n_ptv = 0)
-  expect_equal(obs, exp)
+  # Check case of multiple varaints per annotation category.
```
Comment (ctk3b) [inline tests/testthat/test-counts.R:26]:
```suggestion
  # Check case of multiple variants per annotation category.
```

File: tests/testthat/test-counts.R [hunk 3]
```diff
@@ -11,34 +11,24 @@ test_that("Check variant counter.", {
...
-  
-  # MAC filter of 100: count alleles.
-  obs <- CountAlleles(anno, geno, count_carriers = FALSE, min_mac = 100)
-  exp <- c(n_bmv = 0, n_dmv = 0, n_ptv = 0)
-  expect_equal(obs, exp)
-  
-  # MAC filter of 100: count carriers.
-  obs <- CountAlleles(anno, geno, count_carriers = TRUE, min_mac = 100)
-  exp <- c(n_bmv = 0, n_dmv = 0, n_ptv = 0)
-  expect_equal(obs, exp)
```
Comment (ctk3b) [inline tests/testthat/test-counts.R:42]: n.b. Any particular reason why we dropped these additional test cases?

**Review by ctk3b — APPROVED** [review]
LGTM % some small non-blocking questions/comments

Could you also document the motivation for moving the counts function to C++? Any reasons beyond performance?

File: R/allelic_series_test.R (hunk 1 above)
Comment (zmccaw-insitro) [reply R/allelic_series_test.R:490]: Makes sense, updated as suggested.

File: tests/testthat/test-counts.R (hunk 3 above)
Comment (zmccaw-insitro) [reply tests/testthat/test-counts.R:42]: We don't need to test the `count_carriers` option anymore, as these numbers are shown by default, but I'll add back the `min_mac = 100` test case.

File: tests/testthat/test-counts.R (hunk 2 above)
Comment (zmccaw-insitro) [reply tests/testthat/test-counts.R:26]: Fixed, thanks!

**Comment by zmccaw-insitro** [comment]
> Could you also document the motivation for moving the counts function to C++? Any reasons beyond performance?

Since the counts are now calculated and reported by default, I opted to make the counting function as fast as possible.

**Comment by ctk3b** [comment]
> > Could you also document the motivation for moving the counts function to C++? Any reasons beyond performance?
> 
> Since the counts are now calculated and reported by default, I opted to make the counting function as fast as possible.

Any quick benchmarking worth showing that convinced you that you succeeded? 😁

**Comment by zmccaw-insitro** [comment]
> Any quick benchmarking worth showing that convinced you that you succeeded? 😁

Sure, please see below (testing on 100 SNPs for 1e5 subjects):
Screenshot 2024-07-25 at 1 33 15 PM

**Comment by ctk3b** [comment]
Sweet - LGTM!

**Comment by zmccaw-insitro** [comment]
Thank you!

**Pull request merged**